.github/workflows/build-and-run-batch-job.yaml

```diff
@@ -1,26 +1,58 @@
-# Workflow that builds a Docker image containing the model code,
-# pushes it to the GitHub Container Registry, and then optionally uses
-# that container image to run the model using an AWS Batch job.
+# Reusable workflow that builds a Docker image, pushes it to the GitHub
+# Container Registry, and then optionally uses that container image to run
+# an AWS Batch job.
 #
-# Images are built on every commit to a PR or main branch in order to ensure
-# that the build continues to work properly, but Batch jobs are gated behind
-# a `deploy` environment that requires manual approval from a
-# @ccao-data/core-team member.
-
-name: deploy
+# Requirements:
+#
+#   * A Dockerfile must be defined in the root of the repo whose workflow is
+#     calling this one.
+#   * A `deploy` environment must be configured in the calling repo. This
+#     environment is used to protect the `run` job, which must be approved by
+#     a core team member before it will run.
+#   * The calling workflow must grant the following permissions to the job
+#     that calls this workflow:
+#       * contents: read
+#       * id-token: write
+#       * packages: write
+#   * Various required inputs and secrets, documented below, must be passed in
+#     by the calling workflow.
+
+name: build-and-run-batch-job
 
 on:
-  pull_request:
-  workflow_dispatch:
-  push:
-    branches: [master]
+  workflow_call:
+    inputs:
+      role-duration-seconds:
+        description: How long IAM role used to auth with AWS can be valid.
+        required: false
+        type: string
+        default: 3600
+      vcpu:
+        description: >
+          Count of cVPUs to provision for the container. Per AWS requirements,
+          must be formatted as a float, e.g. 1.0 for 1 vCPU. The minimum
+          is 1 vCPU and values must be specified in increments of 0.25.
+        required: false
+        type: string
+        default: "1.0"
+      memory:
+        description: Count of megabytes of RAM to provision for the container.
+        required: false
+        type: string
+        default: "4096"
+
+    secrets:
+      AWS_IAM_ROLE_TO_ASSUME_ARN:
+        required: true
+      AWS_ACCOUNT_ID:
+        required: true
 
 env:
   DOCKER_REGISTRY: ghcr.io
   DOCKER_IMAGE_NAME: ${{ github.repository }}
 
 jobs:
-  publish-docker-image:
+  build:
```

Comment:
I renamed these jobs for the sake of simplicity, since reusable workflow names are displayed in the GitHub Actions UI under a namespace, e.g. `build-and-run-model / build` rather than just `build`.